server/drivers/hd44780-gpiod.c

```diff
@@ -145,9 +200,10 @@ hd_init_gpiod(Driver *drvthis)
 		return -1;
 	}
 
-	pins = malloc(sizeof(gpio_pins));
+	pins = calloc(1, sizeof(gpio_pins));
 
-	pins->chip = gpiod_chip_open_lookup(chip);
+	snprintf(dev_path, sizeof(dev_path), "/dev/gpiochip%s", chip);
+	pins->chip = gpiod_chip_open(dev_path);
```

Comment:
I think, this changes behavior. But it wasn't explicitly documented behavior, so I hope it is okay.

It's a bit unfortunate though. The old call allowed to pass some unique identifier like `gpio.085f3b00`, that at least some systems have. If we are now back to sequentially indexed devices, then unstable probing order at boot time might actually break things.

Reply:
Yes, I was also thinking how to handle it when I was writing that code. But as you said it is currently undocumented, so I've changed the approach and I think the most flexible for user is to let this be a full path of a gpio device. User may also want to use eg. some UDEV rules (something similar like `/dev/disk/by-id...` - so restricting to `/dev/SOMETHING` is maybe not a good choice).